.github/workflows/tests.yaml

```
@@ -1,25 +1,20 @@
-on: [push, pull_request]
+on: [push, pull request]
 
 jobs:
   tests:
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout repository and submodules
-      uses: actions/checkout@v2
-      with:
-        submodules: recursive
+      - uses: actions/checkout@v3
 
-    - name: Install nix 2.3.6
-      uses: cachix/install-nix-action@v13
-      with:
-        install_url: https://releases.nixos.org/nix/nix-2.3.6/install
-        nix_path: nixpkgs=channel:nixos-unstable
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
 
-    - name: Use maker and dapp cachix
-      uses: cachix/cachix-action@v10
-      with:
-        name: maker
-        extraPullNames: dapp
+      - name: Install Dependencies
+        run: forge install
```

```suggestion
        run: git submodule update --init --recursive
```
I think this way was preferred to install the dependencies

---

I copied the `forge intstall` commad from the D3M repo yaml.
Do you know why is it not good?

---

It should install missing dependencies automatically when testing, so it should not even be needed here, but I guess if we want to specified it the `git` way helps fixing some edge cases I guess.

---

I don't know about this "git submodule update --init --recursive" issue @naszam is bringing up.  I know there was some problems with "forge install" recursing at one point, and it probably still has that issue, but if we can get the repo working with "forge install", I think we should strive for that.

---

maybe adopting a `foundry.toml` could ensure a more compact config settings and eventually address those issues if they arise.

---

but yeah, let's get the CI running and if it passes we can go with that for now

---

It seems working in #37